papertty.py

```diff
@@ -537,6 +582,73 @@ def sigusr1_handler(sig, frame):
                 oldimage = None
                 oldbuff = ''
                 flags['scrub_requested'] = False
+
+            if flags['show_menu']:
+                flags['show_menu'] = False
+                print()
+                print('Rendering paused. Enter')
+                print('    (f) to change font,')
+                print('    (s) to change spacing,')
+                if ptty.is_truetype:
+                    print('    (h) to change font size,')
+                print('    (x) to exit,')
+                print('    anything else to continue.')
+                print('Use --font {} --size {} --spacing {} for the current setting.'.format(ptty.fontfile, ptty.fontsize, ptty.spacing))
+
+                ch = sys.stdin.readline().strip()
+                if ch == 'x':
+                    if not noclear:
+                        ptty.showtext(oldbuff, fill=ptty.white, **textargs)
+                    sys.exit(0)
+                elif ch == 'f':
+                    print('Current font: {}'.format(ptty.fontfile))
+                    font_name = click.prompt('Enter new font (leave empty to abort)')
+                    if font_name:
+                        ptty.spacing = spacing
+                        ptty.font = ptty.load_font(font_name, keep_if_not_found=True)
+                        if autofit:
+                            max_dim = ptty.fit(portrait)
+                            print("Automatic resize of TTY to {} rows, {} columns".format(max_dim[1], max_dim[0]))
+                            ptty.set_tty_size(ptty.ttydev(vcsa), max_dim[1], max_dim[0])
+                        oldbuff = None
+                    else:
+                        print('Font not changed')
+                elif ch == 's':
+                    print('Current spacing: {}'.format(ptty.spacing))
+                    new_spacing = None
+                    try:
+                        new_spacing = int(click.prompt('Enter new spacing (leave empty to abort)'))
+                    except:
+                        pass
+                    if new_spacing or new_spacing == 0:
```

Comment:
Ah, so that's how that works. A bit perverse to have the default value to be of a different type than the one you explicitly specify, but if that's how the library works …

Reply:
Yeah, I guess in a more normal case we could just say "Press Ctrl-C to skip", but this "leave empty to skip" needs a tiny workaround.

Reply:
Ideally, I would listen for a keypress every loop, but that turned out to be surprisingly complicated, at least on Linux. `select.select()`, for example, doesn't seem to trigger until you hit return, so that's a bit inconvenient. But then again, I only see this as a quick and dirty solution anyway.